frontend/app-development/features/processEditor/handlers/OnProcessTaskAddHandler.ts

```diff
@@ -1,26 +1,71 @@
-import type { OnProcessTaskEvent } from '@altinn/process-editor/types/OnProcessTask';
 import { PaymentPolicyBuilder } from '../../../utils/policy';
+import type { OnProcessTaskEvent } from '@altinn/process-editor/types/OnProcessTask';
 import type { Policy } from 'app-shared/types/Policy';
+import type { LayoutSetConfig } from 'app-shared/types/api/LayoutSetsResponse';
+import { getDataTypeIdFromBusinessObject } from '@altinn/process-editor/utils/hookUtils/hookUtils';
+import type { AddLayoutSetMutation } from '../../../hooks/mutations/useAddLayoutSetMutation';
 
 export class OnProcessTaskAddHandler {
   constructor(
     private readonly org: string,
     private readonly app: string,
     private readonly currentPolicy: Policy,
+    private readonly addLayoutSet: AddLayoutSetMutation,
     private readonly mutateApplicationPolicy: (policy: Policy) => void,
+    private readonly addDataTypeToAppMetadata: (data: {
+      dataTypeId: string;
+      taskId: string;
+    }) => void,
   ) {}
 
   /**
    * This handler is responsible to react on task add event, to mutate files that are related to the task, but not a part of process-editor domain.
    * @param taskMetadata
    */
   public handleOnProcessTaskAdd(taskMetadata: OnProcessTaskEvent): void {
+    if (taskMetadata.taskType === 'data') {
+      this.handleDataTaskAdd(taskMetadata);
+    }
+
     if (taskMetadata.taskType === 'payment') {
       this.handlePaymentTaskAdd(taskMetadata);
     }
+
+    if (taskMetadata.taskType === 'signing') {
+      this.handleSigningTaskAdd(taskMetadata);
+    }
+  }
+
+  /**
+   * Adds a layout set to the added data task
+   * @param taskMetadata
+   * @private
+   */
+  private handleDataTaskAdd(taskMetadata: OnProcessTaskEvent): void {
+    this.addLayoutSet(this.createLayoutSetConfig(taskMetadata.taskEvent));
   }
 
+  /**
+   * Adds a dataType, layoutSet and default policy to the added payment task
+   * @param taskMetadata
+   * @private
+   */
   private handlePaymentTaskAdd(taskMetadata: OnProcessTaskEvent): void {
+    // Add layoutSet for the task
```

Comment:
Is this comment needed? Same for line 57 and 68 😄

Reply:
Since this method performs multiple tasks, I added comments to each "thing" to increase readability by creating a "visual section" and clarify what happens within the method.

Reply:
I agree that a comment can be nice when it is easy to get confused by the code, but the next line says `this.addLayoutSet(...`. It might be a personal preference, but I believe that is readable to understand that it adds a layoutset. 😅 What do you think?